### PR TITLE
Document database restore

### PR DESCRIPTION
**Pull request by jontours**
Fairly straightforward instructions so far.  Only for UAA is it somewhat a grey area as to what should be restored.

### CODE REVIEW

**Comment by jontours** [comment]
This is ready for review.  If there are some steps that I'm missing let me know.  I seem to remember when there was the Tock DB corruption that after we performed the restore, we then informed the Tock team to point their app to a new database?  Or something?  That doesn't seem to be necessary from the RDS documentation but wouldn't be surprised if there are wrinkles there.

**Comment by brittag** [comment]
Thanks @jontours! This looks like a good start; I copyedited it a bit. But is it enough information for a platform operator to be able to confidently restore a backup for Cloud Foundry, BOSH, Concourse, or UAA data? After you restore the snapshot, what do you need to do to ensure that the system can use it? Do you need to do anything with credentials for the database? I remember we had to work this out in the [contingency plan exercise](https://docs.google.com/document/d/15u70uk8r_zNMMR8bYHWumNIzSLskwvBHNsEhYiYsaW8/edit#), but I don't remember what the answer was. Looks like a platform squad member should review this next.

**Comment by timothy-spencer** [comment]
Re: tock db corruption

I just restored a db from a snapshot before the sadness happened, told them what hostname to use, and they were able to log in and do some sort of schema/data extraction that fixed the problem themselves.

**Review by timothy-spencer — COMMENTED** [review]
I would like to echo Britta's request for a bit more documentation on how to handle recovering the databases for the service (bosh/atc, maybe domains broker db) up into a state where they are being used by the services that they serve.

I'm not entirely sure what that process is, but perhaps it could be doable to sorta one-off a terraform run like what they do here:  https://medium.com/@vankhoa011/how-i-use-terraform-to-restore-the-latest-snapshot-from-productions-db-to-staging-s-db-aws-rds-6ad4f6620df2

By "one-off", I mean that maybe you could figure out a way to do a one-off run of the terraform stuff in cg-provision where the stack had the proper "snapshot_identifier" in it so that it would launch a new db to replace the (presumably missing) one and pull in the snapshot from the old db.  You might need to push the resulting state file up to the tfstate bucket after that.  Not 100% sure.

Either that, or your procedure could be to restore from snapshot/PITR, then dump/load the data into the stuff created by cg-provision.

**Comment by timothy-spencer** [comment]
Not sure if it helps, but it looks like the uaa stuff is in the bosh db, so if you restore the bosh db properly, uaa should be recovered as well.

**Comment by sharms** [comment]
Hi folks - the scope of this story is to document our current process.  Lets open up a new ticket for planned changes or improvements and not block on this one.

**Review by sharms — DISMISSED** [review]

**Comment by timothy-spencer** [comment]
I'm confused.  This document is supposed to contain our current process, agreed.  But our current process includes being able to restore our service from backups, no?  If that's not true, then OK, but it was my understanding that this was the core of what we were trying to document.

**Comment by mogul** [comment]
Per review this morning, we want to include a bit at the end here about what to do with the restored DB: import it into terraform, then adjust the secrets to point to it.

**Comment by timothy-spencer** [comment]
Should I be trying to add this stuff to the process, or should I just wait for updated content to review?

**Comment by jontours** [comment]
I'm currently working on adding documentation on using concourse to perform a terraform refresh for the final step of updating the environment for a restored DB.  Will probably be opening up a PR in `cg-provision` today.

File: content/docs/ops/runbook/restoring-rds.md
Comment (jmcarp) [inline content/docs/ops/runbook/restoring-rds.md]: This applies to more than bosh and concourse--this is the process for any platform rds instance. Maybe something like "Platform applications"?

File: content/docs/ops/runbook/restoring-rds.md
Comment (jmcarp) [inline content/docs/ops/runbook/restoring-rds.md]: Concourse might be unavailable for other reasons. WDYT about dropping the parenthetical here?

File: content/docs/ops/runbook/restoring-rds.md
Comment (jmcarp) [inline content/docs/ops/runbook/restoring-rds.md]: "These instructions are augment" should maybe be "These instructions augment". And "documenation" should be "documentation".

File: content/docs/ops/runbook/restoring-rds.md
Comment (jmcarp) [inline content/docs/ops/runbook/restoring-rds.md]: We might want to add a note about instance types, networks, security groups, etc. We should make sure the new instance lives in the same subnet group, has the same security groups, etc., as the original.

File: content/docs/ops/runbook/restoring-rds.md
Comment (jmcarp) [inline content/docs/ops/runbook/restoring-rds.md]: If the new instance has the same credential as the old one, do we need to send the credentials to the user again?

File: content/docs/ops/runbook/restoring-rds.md
Comment (timothy-spencer) [reply content/docs/ops/runbook/restoring-rds.md]: I think good things, myself.

File: content/docs/ops/runbook/restoring-rds.md
Comment (jontours) [reply content/docs/ops/runbook/restoring-rds.md]: I didn't make this edit but can remove it.

Comment (sharms) [reply content/docs/ops/runbook/restoring-rds.md]: This specific section was how we did it previously, it may not apply now

File: content/docs/ops/runbook/restoring-rds.md
Comment (timothy-spencer) [inline content/docs/ops/runbook/restoring-rds.md]: I really think that VPC is super important to call out here too.

File: content/docs/ops/runbook/restoring-rds.md
Comment (jontours) [reply content/docs/ops/runbook/restoring-rds.md]: Well, it may actually.  At the very least a tenant will have to `update-user-provided-service`.  But if you run update, according to CF docs, any parameters not provided in the update will be deleted.  So I think these instructions are valid.

File: content/docs/ops/runbook/restoring-rds.md
Comment (jontours) [reply content/docs/ops/runbook/restoring-rds.md]: Instructions added taking note of these settings.

File: content/docs/ops/runbook/restoring-rds.md
Comment (jontours) [reply content/docs/ops/runbook/restoring-rds.md]: It is and it's noted in the following paragraph.  Remember we couldn't set it until after restoring the snapshot?

File: content/docs/ops/runbook/restoring-rds.md
Comment (jmcarp) [reply content/docs/ops/runbook/restoring-rds.md]: Why will tenants need to update the service?

Comment (jontours) [reply content/docs/ops/runbook/restoring-rds.md]: In the case that the host name changes.  For example I thought the Tock team had to do an update, but I think you could do the restore in a way that the host name doesn't change so I can remove these steps.

Comment (jmcarp) [reply content/docs/ops/runbook/restoring-rds.md]: But the tenant won't/shouldn't be using the user-provided service to provide credentials to applications. AFAICT the user-provided service is just a secure way for us to send credentials to users so we don't have to send them over email or something.

File: content/docs/ops/runbook/restoring-rds.md
Comment (timothy-spencer) [reply content/docs/ops/runbook/restoring-rds.md]: VPC could (and should) be set, but the security groups could not be set upon restore.  The security groups are what needed to be updated, not the VPC.

Comment (jontours) [reply content/docs/ops/runbook/restoring-rds.md]: You're right!  added.

**Comment by jontours** [comment]
If there are any further edits requested let me know, otherwise I believe this is done.  Also verified the formatting looks good by deploying locally.

**Review by timothy-spencer — DISMISSED** [review]
Yay!

**Review by brittag — APPROVED** [review]
This looks sufficient to me, thanks everyone!

**Pull request merged**
